platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java

```diff
@@ -537,7 +542,10 @@ private final BundleStateSet getCurrentBundleStates(Set<Feature> features) {
 
     for (Feature curFeature : features) {
       for (BundleInfo curBundleInfo : curFeature.getBundles()) {
-        Bundle curBundle = getContext().getBundle(curBundleInfo.getLocation());
+        BundleContext context = getContext();
+        Bundle curBundle =
+            (context != null) ? context.getBundle(curBundleInfo.getLocation()) : null;
```

Comment:
✏️  Kind of weird that we are setting curBundle to null just to check it again on the next line? Can we re-work this logic?

Plus another case where the double for loops don't seem to be doing anything if context is null. I am with Chris on maybe not having getContext() return null in the first place. Especially if that should never happen in the first place anyway.